assets/volumedatasourcevalidator/deployment.yaml

```diff
@@ -0,0 +1,32 @@
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  name: volume-data-source-validator
+  namespace: openshift-cluster-storage-operator
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: volume-data-source-validator
+  template:
+    metadata:
+      labels:
+        app: volume-data-source-validator
+    spec:
+      serviceAccountName: volume-data-source-validator
+      containers:
+        - name: volume-data-source-validator
+          image: ${VOLUME_DATA_SOURCE_VALIDATOR_IMAGE}
+          args:
+            - "--v=${LOG_LEVEL}"
+            - "--http-endpoint=:8080"
+            - "--leader-election=true"
+            - "--leader-election-lease-duration=137s"
+            - "--leader-election-renew-deadline=107s"
+            - "--leader-election-retry-period=26s"
+            - "--leader-election-namespace=openshift-cluster-storage-operator"
+          imagePullPolicy: Always
+          ports:
+            - containerPort: 8080
+              name: http-endpoint
+              protocol: TCP
```

Comment:
What is port 8080 used for? You'll need a NetworkPolicy for that too. If it's metrics, try removing `--http-endpoint=:8080` so it will default to `8443` and then set `containerPort: 8443` and add the `openshift.storage.network-policy.operator-metrics: allow` annotation.

If it is something other than metrics and it really has to be port 8080, then you'll need a new NetworkPolicy + annotation to allow ingress for that port.

Reply:
Ah! Okay, it looks like it should be accessible via kube-rbac-proxy in your latest changes.